inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/experimental_detectron_roi_feature_extractor_ref.cl

```diff
@@ -0,0 +1,130 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "include/batch_headers/common.cl"
+#include "include/batch_headers/data_types.cl"
+
+DEFINE_SPATIAL_SCALES;
+DEFINE_LEVEL_SIZES;
+
+inline int FUNC(get_pyramid_level_for_roi)(const __global INPUT0_TYPE* current_roi) {
+    const INPUT0_TYPE canonical_scale = 224.0;
+    const int canonical_level = 2;
+
+    int result = NUM_PYRAMID_LEVELS;
+
+    const INPUT0_TYPE x0 = current_roi[0];
+    const INPUT0_TYPE y0 = current_roi[1];
+    const INPUT0_TYPE x1 = current_roi[2];
+    const INPUT0_TYPE y1 = current_roi[3];
+
+    const INPUT0_TYPE area = (x1 - x0) * (y1 - y0);
+    if (area > 0) {
+        result = (int)round(canonical_level + log2(sqrt(area) / canonical_scale));
+        result = max(0, min(result, NUM_PYRAMID_LEVELS - 1));
+    }
+    return result;
+}
+
+KERNEL(experimental_detectron_roi_feature_extractor_ref)(const __global INPUT0_TYPE* src_rois,
+                                                         INPUT_LEVEL_PARAMS,
+                                                         __global OUTPUT_TYPE* dst_data)
+{
+    const size_t oxy = get_global_id(0);
```

Comment:
It's better to use `uint/int` data types for `get_global_id()` return value, since further calculation will use `size_t` datatype for calculations (`oxy % POOLED_WIDTH` and `oxy / POOLED_WIDTH`) that is slower than `uints` calculation

Reply:
done